### PR TITLE
Formplayer post message api

### PR DESCRIPTION
**Pull request by benrudolph**
@biyeun @wpride this provides us with a framework to send messages to app preview from HQ. right now the use case is for providing a back button that does not live inside of app preview. in the future we'll be able to do things like tell app preview when it's a version behind the version of the app loaded in the app manager or send cool signals from vellum to live preview. the API is pretty simple right now: `{ action: <action> }`. in the future though i'd imagine we'd have the ability to pass parameters etc. 

@biyeun this doesn't implement a styled back button as discussed. fine to merge, but should style it before we release. the button is in the very very bottom left corner haha

screen shot 2016-10-26 at 5 22 19 pm

cc: @nickpell @emord

### CODE REVIEW

**Comment by dimagimon** [comment]
There are 23 errors:

* corehq/apps/app_manager/static/app_manager/js/preview_app.js, line 55 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 40 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 41 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 42 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 56 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 58 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 1 - 'describe' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 2 - 'describe' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 3 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 4 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 8 - 'beforeEach' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 12 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 14 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 17 - 'afterEach' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 18 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 21 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 29 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 35 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 40 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 46 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 57 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 63 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 64 - 'FormplayerFrontend' is not defined. (no-undef)

**Comment by dimagimon** [comment]
There are 23 errors:

* corehq/apps/app_manager/static/app_manager/js/preview_app.js, line 55 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 40 - Expected indentation of 12 space characters but found 16. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 41 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 42 - Expected indentation of 16 space characters but found 20. (indent)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 56 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/hq.events.js, line 58 - Missing semicolon. (semi)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 1 - 'describe' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 2 - 'describe' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 3 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 4 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 8 - 'beforeEach' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 12 - Missing trailing comma. (comma-dangle)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 14 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 17 - 'afterEach' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 18 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 21 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 29 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 35 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 40 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 46 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 57 - 'it' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 63 - 'FormplayerFrontend' is not defined. (no-undef)
* corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js, line 64 - 'FormplayerFrontend' is not defined. (no-undef)

File: corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        var Receiver = FormplayerFrontend.HQ.Events.Receiver,
+            Actions = FormplayerFrontend.HQ.Events.Actions,
+            origin = 'myorigin',
+            triggerSpy,
+            dummyEvent;
+        beforeEach(function() {
+            triggerSpy = sinon.spy();
+            dummyEvent = {
+                origin: origin,
+                data: {}
```
Comment (dimagimon) [inline corehq/apps/cloudcare/static/cloudcare/js/formplayer/spec/hq.events_spec.js]: Missing trailing comma. (comma-dangle)

**Comment by wpride** [comment]
I'm will pride and I approve this message

**Comment by benrudolph** [comment]
self merge based on approval

**Pull request merged**